### ISSUE TITLE
Support running a script for certain events.

### ISSUE THREAD

**Issue by eepykate** [unixporn/trup#167]
### But why?

Can make it easier & faster to deal with stuff such as raids,  
just a general improvement in bot usability & versatility,  
makes a more sophisticated blocklist easy to achieve as a sideaffect.

### Supported variables
 - channel
 - message (lowercased content of message)
 - messageid
 - user   (snowflake)
 - roles  (snowflakes separated by space)
 - join   (snowflake of join date)

### Supported events:
 - `user-join`
 - `new-message`

`message-delete` / `message-edit` don't sound too useful to me.

### Supported actions: 
 - mute
   * `mute <duration> [message]`
 - message
   * `message <channel/user ID> <contents>`   (*can* be hardcoded, don't restrict to the defined variables)
 - modping
   * `modping [reason]`
 - delete
   * simply deletes the matched message
 - purge
   * `purge <amount/duration>`  (deletes \<x\> messages from user)

Any combination of these should be possible.

### Example

```
!script add new-message `​``bash
if [[ "$message" =~ 'lineageos is good' ]] &&
# only old accounts >:)
[[ "$user" -lt 729492634438270977 ]]; then
  mute 2h having the wrong opinion
fi
`​``
```

Bash might be a tad too inefficient for this, but it's what I know so it's what my example is in.  
Another language (such as `lua`) might be worth considering, since supporting multiple langs sounds messy & time consuming (especially if we care about feature parity).

**Comment by 6gk** [comment]
i was probably forgetting a user-join event.

**Comment by elkowar** [comment]
not a big fan honestly.
Wrapping the relevant parts of discordgo to just have handling those events be more convenient on a go level seems viable, and would make this less,... dangerous. the required basics of go aren't that much harder than basic shell scripting, so that shouldn't be an issue.

**Comment by eepykate** [comment]
@elkowar So you're suggesting making the command take go as input rather than translating from another lang?  This honestly didn't even cross my mind lol.

**Comment by elkowar** [comment]
no, just,... write the code and commit lol. I find the idea of having on-the-spot created, not reviewed scripts that have the power to mute veeeeery scary

**Comment by eepykate** [comment]
fair enough ig.
